### PR TITLE
[INLONG-6045][Sort] Support all migrate for Oracle connector

### LINKED ISSUE #6045: [Feature][Sort] Support all migrate for Oracle connector
### Description

 Sort support all migrate for Oracle connector

### Use case

null

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by e-mhui**
*Sort support all migrate for Oracle connector*

- Fixes #6045

### Motivation

The thing is sometime user want to export all data from database with diffrent schema to kafka with canal json
This is something flinkcdc cannot do since we all need to specify the source shema when writing sql

### Modifications

refer: #4010

### Documentation

  - Does this pull request introduce a new feature? (yes)

### Verifying this change

run `AllMigrateOracleTest`

### How to config option

| Option | Required | Default | Type | Description |
| ---| ---| ---| ---| ---|
|source.multiple.enable|optional| false|String|Specify `'source.multiple.enable' = 'true'` to enable the all database migration function | 
|schema-name|required|(none)|String| Schema name of the Oracle database to monitor. If you want to capture multiple schemas, you can use commas to separate them. For example: `'schema-name' = 'SCHEMA1,SCHEMA2'` |
|table-name| required | (none) |String| Table name of the Oracle database to monitor. If you want to capture multiple tables, you can use commas to separate them. For example: `'table-name' = 'SCHEMA1.TB.*, SCHEMA2.TB1'`|

### CODE REVIEW

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSource.java [hunk 1]
```diff
@@ -130,13 +129,14 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
...
                                 OracleDeserializationConverterFactory.instance())
+                        .setMigrateAll(migrateAll)
                         .build();
         OracleSource.Builder<RowData> builder =
                 OracleSource.<RowData>builder()
                         .hostname(hostname)
                         .port(port)
                         .database(database)
-                        .tableList(schemaName + "." + tableName)
+                        .tableList(tableName)
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSource.java:139]: manager should change the parameter rules noted

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/internal/DebeziumChangeConsumer.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.Map;
+import org.apache.flink.annotation.Internal;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Consume debezium change events. */
+@Internal
+public class DebeziumChangeConsumer
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/internal/DebeziumChangeConsumer.java:35]: the classes copied from flink cdc should be noted in the license file

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/OracleValidator.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.inlong.sort.cdc.oracle.debezium.Validator;
+import org.apache.inlong.sort.cdc.oracle.util.OracleJdbcUrlUtils;
+
+/** Validates the version of the database connecting to. */
+public class OracleValidator implements Validator {
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/OracleValidator.java:31]: can such class be replaced by the dependency provided by the flink cdc ?

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSource.java (hunk 1 above)
Comment (e-mhui) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSource.java:139]: Manager does not need to change, the previous rule of 'table name' was'<tableName>'. Now it needs to be changed to'<schemaName>.<tableName>`.

File: licenses/inlong-sort-connectors/LICENSE [hunk 4]
```diff
@@ -489,6 +489,30 @@
...
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/utils/DatabaseHistoryUtil.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/utils/TemporalConversions.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/DebeziumDeserializationSchema.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/DebeziumSourceFunction.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/JsonDebeziumDeserializationSchema.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/StringDebeziumDeserializationSchema.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleDeserializationConverterFactory.java
+       inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleReadableMetaData.java
```
Comment (EMsnap) [inline licenses/inlong-sort-connectors/LICENSE:515]: oracle copied from elasticsearch?

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/internal/DebeziumChangeConsumer.java (hunk 2 above)
Comment (e-mhui) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/internal/DebeziumChangeConsumer.java:35]: done.

File: licenses/inlong-sort-connectors/LICENSE (hunk 4 above)
Comment (e-mhui) [reply licenses/inlong-sort-connectors/LICENSE]: done.

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/AllMigrateOracleTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.inlong.sort.protocol.transformation.relation.NodeRelation;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AllMigrateOracleTest {
+
+    private OracleExtractNode buildAllMigrateExtractNode() {
+        List<FieldInfo> fields = Arrays.asList(
+                new MetaFieldInfo("data", MetaField.DATA));
+        Map<String, String> option = new HashMap<>();
```
Comment (gong) [inline inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/AllMigrateOracleTest.java:52]: Does It need support MetaField.DATA_BYTES?@e-mhui@EMsnap

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java [hunk 6]
```diff
@@ -95,6 +95,12 @@ public class OracleTableSourceFactory implements DynamicTableSourceFactory {
                             "Optional startup mode for Oracle CDC consumer, valid enumerations are "
                                     + "\"initial\", \"latest-offset\"");
 
+    public static final ConfigOption<Boolean> MIGRATE_ALL =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java:98]: Maybe it is can also extracted to class Constants of the mode 'base'?

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java [hunk 7]
```diff
@@ -95,6 +95,12 @@ public class OracleTableSourceFactory implements DynamicTableSourceFactory {
...
+            ConfigOptions.key("migrate-all")
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java:99]: Maybe it is better to call 'source.multiple.enable'?

Comment (e-mhui) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java]: done.

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java (hunk 6 above)
Comment (e-mhui) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/table/OracleTableSourceFactory.java]: done.

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java [hunk 8]
```diff
@@ -130,6 +130,12 @@ public final class Constants {
                             + "is used extract database name from the raw binary data, "
                             + "this is only used in the multiple sink writing scenario.");
 
+    public static final ConfigOption<Boolean> SOURCE_MULTIPLE_ENABLE =
+            ConfigOptions.key("source.multiple.enable")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Whether migrate multiple databases");
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java:137]: Maybe it is better to call 'Whether enable multiple source'?

Comment (e-mhui) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java:137]: Now there is only one source, It should be appropriate to Call 'Which enable migrate multiple databases'.

File: inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/JsonDebeziumDeserializationSchema.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+import org.apache.kafka.connect.json.JsonConverterConfig;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.ConverterConfig;
+import org.apache.kafka.connect.storage.ConverterType;
+
+/**
+ * A JSON format implementation of {@link DebeziumDeserializationSchema} which deserializes the
+ * received {@link SourceRecord} to JSON String.
+ */
+public class JsonDebeziumDeserializationSchema implements DebeziumDeserializationSchema<String> {
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/JsonDebeziumDeserializationSchema.java:37]: are JsonDebeziumDeserializationSchema and StringDebeziumDeserializationSchema ever used?

Comment (e-mhui) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/java/org/apache/inlong/sort/cdc/oracle/debezium/JsonDebeziumDeserializationSchema.java]: done.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/Metadata.java [hunk 10]
```diff
@@ -54,6 +54,10 @@ default String getMetadataKey(MetaField metaField) {
             case OP_TS:
                 metadataKey = "op_ts";
                 break;
+            case DATA:
+            case DATA_BYTES:
+                metadataKey = "meta.data";
```
[the remark this answers is not in the document]
Comment (e-mhui) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/Metadata.java:59]: done.

**Review by thesumery — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**